src/db.c

```diff
@@ -258,7 +258,7 @@ int getKeySlot(sds key) {
      * so we must always recompute the slot for commands coming from the primary.
      */
     if (server.current_client && server.current_client->slot >= 0 && server.current_client->flag.executing_command &&
-        !server.current_client->flag.primary) {
+        !server.current_client->flag.cross_slot && !server.current_client->flag.primary) {
```

Comment:
We've used `->slot = -1` to indicate it's cross slot in the past. We could also reserve `-2` for this purpose. This allows us to avoid taking another flag.